// @chakra-ui/vue/dist/esm/CCode.js

/*
 * ! @chakra-ui/vue v0.10.2 by Jonathan Bakebwa @codebender828
 * MIT License
 * https://github.com/chakra-ui/chakra-ui-vue
 */

import { R as createStyledAttrsMixin, u as useVariantColorWarning } from './index-88b4bf19.js';
import './index-980f0b61.js';
import '@emotion/css';
import 'vue';
import './color-mode-observer-5bf3a66d.js';
import { a as SNA } from './props.types-26e3a1e7.js';
import { u as useBadgeStyle } from './badge.styles-02c8e304.js';

/**
 * Hey! Welcome to @chakra-ui/vue Code
 *
 * Code is a component used to display inline code.
 *
 * @see Docs     https://vue.chakra-ui.com/code
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue/blob/master/packages/chakra-ui-core/src/CCode/CCode.js
 */
/**
 * CCode component
 *
 * Inline code component
 *
 * @extends CBox
 * @see Docs https://vue.chakra-ui.com/code
 */

var CCode = {
  name: 'CCode',
  mixins: [createStyledAttrsMixin('CCode')],
  props: {
    variantColor: {
      type: String,
      default: 'gray'
    },
    fontFamily: {
      type: SNA,
      default: 'mono'
    }
  },
  computed: {
    badgeStyle: function badgeStyle() {
      useVariantColorWarning(this.theme, 'CCode', this.variantColor);
      return useBadgeStyle({
        variant: 'subtle',
        color: this.variantColor,
        colorMode: this.colorMode,
        theme: this.theme
      });
    },
    componentStyles: function componentStyles() {
      return Object.assign({}, {
        display: 'inline-block',
        fontSize: 'sm',
        px: '0.2em',
        fontFamily: 'mono',
        rounded: 'sm'
      }, this.badgeStyle);
    }
  },
  render: function render(h) {
    return h('code', {
      class: this.className,
      attrs: this.computedAttrs
    }, this.$slots.default);
  }
};

export default CCode;
