// @chakra-ui/vue/dist/cjs/CBadge.js

/*
 * ! @chakra-ui/vue v0.10.2 by Jonathan Bakebwa @codebender828
 * MIT License
 * https://github.com/chakra-ui/chakra-ui-vue
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('./index-a67b3d6b.js');
require('./index-1b221f99.js');
require('@emotion/css');
require('vue');
require('./color-mode-observer-fffa3a3b.js');
var badge_styles = require('./badge.styles-ee2532b9.js');

/**
 * Hey! Welcome to @chakra-ui/vue Badge
 *
 * The Badge component is used to highlight an item's status for quick recognition.
 *
 * @see Docs     https://vue.chakra-ui.com/badge
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue/blob/master/packages/chakra-ui-core/src/CBadge/CBadge.js
 */
/**
 * CBadge component
 *
 * Used to highlight an item's status for quick recognition.
 *
 * @see Docs https://vue.chakra-ui.com/badge
 */

var CBadge = {
  name: 'CBadge',
  mixins: [utils.createStyledAttrsMixin('CBadge')],
  props: {
    variant: {
      type: String,
      default: 'subtle'
    },
    variantColor: {
      type: String,
      default: 'gray'
    },
    as: {
      type: String,
      default: 'div'
    }
  },
  computed: {
    badgeStyles: function badgeStyles() {
      return badge_styles.useBadgeStyle({
        theme: this.theme,
        colorMode: this.colorMode,
        color: this.variantColor,
        variant: this.variant
      });
    },
    componentStyles: function componentStyles() {
      return Object.assign({}, {
        d: 'inline-block',
        textTransform: 'uppercase',
        fontSize: 'xs',
        fontFamily: 'body',
        px: 1,
        rounded: 'sm',
        fontWeight: 'bold',
        whiteSpace: 'nowrap',
        verticalAlign: 'middle'
      }, this.badgeStyles);
    }
  },
  render: function render(h) {
    return h(this.as, {
      class: this.className,
      attrs: this.computedAttrs
    }, this.$slots.default);
  }
};

exports.default = CBadge;
